packages/ui/src/bounty/modals/ContributeFundsModal/ContributeFundsModal.tsx

```diff
@@ -0,0 +1,223 @@
+  const {
+    modalData: { bounty },
+    hideModal,
+    showModal,
+  } = useModal<BountyContributeFundsModalCall>()
+  const { api, isConnected } = useApi()
+  const minFundingLimit = api?.consts.bounty.minFundingLimit.toNumber() ?? 0
+  const { active: activeMember } = useMyMemberships()
+  const { allAccounts } = useMyAccounts()
+  const [amount, setAmount] = useNumberInput(6, minFundingLimit)
+  const [state, send] = useMachine(contributeFundsMachine)
+  const [account, setAccount] = useState<Account>()
+  const balance = useBalance(account?.address)
+
+  const setStakingAmount = useCallback((e: ChangeEvent<HTMLInputElement>) => setAmount(e.target.value), [])
+
+  const valid = useMemo(() => new BN(amount).gten(minFundingLimit) && !!account, [amount, account])
+
+  const setMaxAmount = useCallback(() => {
+    balance && setAmount(balance.transferable.toString())
+  }, [balance])
+
+  const setHalfAmount = useCallback(() => {
+    balance && setAmount(balance.transferable.divn(2).toString())
+  }, [balance])
+  const transaction = useMemo(() => {
+    if (api && isConnected && activeMember) {
+      return api.tx.bounty.fundBounty({ Member: activeMember.id }, bounty.id, amount)
+    }
+  }, [JSON.stringify(activeMember), isConnected])
+
+  const fee = useTransactionFee(activeMember?.controllerAccount, transaction)
+
+  const contribution = useMemo(() => new BN(amount), [amount])
+
+  const nextStep = useCallback(() => {
+    send('NEXT')
+  }, [])
+
+  useEffect(() => {
+    balance &&
+      setAmount(balance.transferable.gten(minFundingLimit) ? String(minFundingLimit) : balance.transferable.toString())
+  }, [balance?.transferable.toString(), account?.address])
+
+  useEffect(() => {
+    if (state.matches(ContributeFundStates.requirementsVerification)) {
+      if (!activeMember) {
+        showModal<SwitchMemberModalCall>({ modal: 'SwitchMember' })
+      } else {
+        nextStep()
+      }
+    }
+  }, [state, activeMember?.id])
+
+  if (!activeMember || !transaction || state.matches(ContributeFundStates.requirementsVerification)) {
+    return null
+  }
+
+  if (state.matches(ContributeFundStates.success)) {
+    return <SuccessModal onClose={hideModal} />
+  }
+
+  if (state.matches(ContributeFundStates.error)) {
+    return (
+      <FailureModal onClose={hideModal} events={state.context.transactionEvents}>
+        {t('modals.contribute.error')}
+      </FailureModal>
+    )
+  }
+
+  if (state.matches(ContributeFundStates.cancel)) {
+    return <FailureModal onClose={hideModal}>{t('common:modals.transactionCanceled')}</FailureModal>
+  }
+
+  if (state.matches(ContributeFundStates.transaction)) {
+    const service = state.children.transaction
+    const controllerAccount = accountOrNamed(allAccounts, activeMember.controllerAccount, 'Controller Account')
+
+    return (
+      <AuthorizeTransactionModal
+        onClose={hideModal}
+        transaction={transaction}
+        service={service}
```

Comment:
Hook for transaction will always return the it is not ready, because it won't reach `prepare` state. You have to pass `state.children['transaction']` as service for it to work.

Reply:
mb, didn't see redeclaration few lines earlier